### ISSUE TITLE
Compatibility problem with VimTeX and coc.nvim

### ISSUE THREAD

**Issue by kaizokuroo** [neoclide/coc.nvim#3694]
## Result from CocInfo

```vim
## versions

vim version: VIM - Vi IMproved 8.2 8024464
node version: v17.7.0
coc.nvim version: 0.0.80-abd8bcc9 2022-03-10 23:58:41 +0800
coc.nvim directory: /home/kaiser/.vim/plugged/coc.nvim
term: dumb
platform: linux

## Log of coc.nvim

2022-03-11T00:17:02.224 INFO (pid:20017) [plugin] - coc.nvim initialized with node: v17.7.0 after 67ms
2022-03-11T00:17:02.226 ERROR (pid:20017) [node-client] - request error on "nvim_eval" [ 'vimtex#re#deoplete' ] Vim(return):E121: Undefined variable: vimtex#re#deoplete Error
    at H1.request (/home/kaiser/.vim/plugged/coc.nvim/build/index.js:29:43394)
    at H1.eval (/home/kaiser/.vim/plugged/coc.nvim/build/index.js:32:3435)
    at activate (/home/kaiser/.config/coc/extensions/node_modules/coc-vimtex/src/index.js:10:24)
    at active (/home/kaiser/.config/coc/extensions/node_modules/coc-vimtex/src/index.js:78:5)
    at Object.exports.activate (/home/kaiser/.config/coc/extensions/node_modules/coc-vimtex/src/index.js:82:7)
    at /home/kaiser/.vim/plugged/coc.nvim/build/index.js:233:1411
    at new Promise (<anonymous>)
    at Object.activate (/home/kaiser/.vim/plugged/coc.nvim/build/index.js:233:1369)
    at dK.activate (/home/kaiser/.vim/plugged/coc.nvim/build/index.js:230:7425)
    at /home/kaiser/.vim/plugged/coc.nvim/build/index.js:232:283
2022-03-11T00:17:02.227 WARN (pid:20017) [workspace] - workspace.showMessage is deprecated, please use window.showMessage instead. 
    at activate (/home/kaiser/.config/coc/extensions/node_modules/coc-vimtex/src/index.js:12:15)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
2022-03-11T00:17:05.977 INFO (pid:20017) [attach] - receive notification: showInfo []
```

## Describe the bug

Hello, so I am very new, it is my second post on Github (I already posted this issue on lervag/VimTeX's page and he redirected me here).

The main problem is this error message (see screenshot) that pops up each time I open a `.tex` file. The message says that VimTeX isn't loaded but the command `:VimtexInfo` works and even the VimTeX compiling mechanism, so it seems that VimTeX isn't the issue here. I have installed `coc.nvim` and `VimTeX` through `plug_vim` and `coc-vimtex` through `CocInstall`.

In the output of `:CocInfo` I noticed that apparently there is a problem with `vimtex#re#deoplete` but I don't even have deoplete installed. Thus I don't quite understand where the problem is.

There might be a problem with the installation path since my VimTeX is only installed in `~/vim/plugged/vimtex` but `coc.nvim` is installed in both path `plugged/` and `bundle/`. Maybe this could be the problem(?)

## Reproduce the bug

**We will close your issue when you don't provide minimal vimrc and we can't
reproduce it**

- Create file `mini.vim` with：

```vim
set encoding=utf-8
set nocompatible
set runtimepath^=/home/kaiser/.vim/plugged/coc.nvim
set runtimepath^=/home/kaiser/.vim/plugged/vimtex
filetype plugin indent on
syntax on
set hidden
" coc-vimtex
let g:coc_global_extensions = ['coc-vimtex']
```

- Create a minimal `test.tex` file with：

```latex
\documentclass{minimal}
\begin{document}
Hello world!
\end{document}
```

- Start vim with command: `vim -u mini.vim test.tex`

- Operate vim.

## Screenshots (optional)

![1646954161 vim](https://user-images.githubusercontent.com/58403918/157771884-587ff4a2-2f63-4a19-9c68-51bcbdfbbb6c.png)

**Comment by chemzqm** [comment]
Make sure you have `:echo g:loaded_vimtex` to have `1`, or you can disable current filetype by configuration `coc.source.vimtex.filetypes`

**Comment by kaizokuroo** [comment]
Thank you for your reply.

`:echo g:loaded_vimtex` gives actually `1` back so this is okay. I tried changing the `coc.source.vimtex.filetyles` to only `["tex"]` in my configuration file and the error message has disappeared.

The problem is now that the auto completion still doesn't work, even without any error message. When I start writing a LaTeX command (e.g. `\usepackage`) the coc prompt doesn't give me any result.

Just in case I give you my `:CocInfo` result
```vim
## versions

vim version: VIM - Vi IMproved 8.2 8024464
node version: v17.7.1
coc.nvim version: 0.0.80-a554a31c 2022-03-11 19:04:48 +0800
coc.nvim directory: /home/kaiser/.vim/plugged/coc.nvim
term: dumb
platform: linux

## Log of coc.nvim

2022-03-11T14:12:49.323 INFO (pid:8132) [plugin] - coc.nvim initialized with node: v17.7.1 after 80ms
2022-03-11T14:15:57.341 INFO (pid:8132) [attach] - receive notification: showInfo []
```

And my config file looks like this:
```json
{
	"coc.source.vimtex.filetypes": ["tex"]
}
```

**Comment by chemzqm** [comment]
Try `"coc.source.vimtex.filetypes": ["tex", "latex"]`

**Comment by kaizokuroo** [comment]
By adding `"latex"` in this setting I get another error message:
```vim
[coc.nvim]:     at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

And here is my `:CocInfo` again:
```vim
## versions

vim version: VIM - Vi IMproved 8.2 8024464
node version: v17.7.1
coc.nvim version: 0.0.80-a554a31c 2022-03-11 19:04:48 +0800
coc.nvim directory: /home/kaiser/.vim/plugged/coc.nvim
term: dumb
platform: linux

## Log of coc.nvim

2022-03-11T21:29:37.905 INFO (pid:4204) [plugin] - coc.nvim initialized with node: v17.7.1 after 76ms
2022-03-11T21:29:37.908 ERROR (pid:4204) [node-client] - request error on "nvim_eval" [ 'vimtex#re#deoplete' ] Vim(return):E121: Undefined variable: vimtex#re#deoplete Error
    at H1.request (/home/kaiser/.vim/plugged/coc.nvim/build/index.js:29:43394)
    at H1.eval (/home/kaiser/.vim/plugged/coc.nvim/build/index.js:32:3435)
    at activate (/home/kaiser/.config/coc/extensions/node_modules/coc-vimtex/src/index.js:19:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
2022-03-11T21:29:37.910 ERROR (pid:4204) [server] - unhandledRejection  Promise {
  <rejected> Error
      at H1.request (/home/kaiser/.vim/plugged/coc.nvim/build/index.js:29:43394)
      at H1.eval (/home/kaiser/.vim/plugged/coc.nvim/build/index.js:32:3435)
      at activate (/home/kaiser/.config/coc/extensions/node_modules/coc-vimtex/src/index.js:19:26)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
} Error
    at H1.request (/home/kaiser/.vim/plugged/coc.nvim/build/index.js:29:43394)
    at H1.eval (/home/kaiser/.vim/plugged/coc.nvim/build/index.js:32:3435)
    at activate (/home/kaiser/.config/coc/extensions/node_modules/coc-vimtex/src/index.js:19:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
2022-03-11T21:29:59.882 INFO (pid:4204) [attach] - receive notification: showInfo []
```

**Comment by chemzqm** [comment]
Seems your vimtex is broken.

**Comment by kaizokuroo** [comment]
So I tried uninstalling and deleting all vim related files `~/.vim`, `~/.config/coc/` etc and tried a fresh install. Nothing changed, the problem is still the same.

Could using gvim be the problem? I use gvim in terminal.

What do you mean by *broken*? I thought that do a fresh install could solve this but now I don't know what to do. Do you think the problem is in VimTeX's side?

**Comment by lervag** [comment]
First, @kaizokuroo: I tried to reproduce your problem, but at first I was not able to. I removed my coc-vimtex folder and installed it fresh, and now I can reproduce your problem. It seems to be introduced a newer version of coc-vimtex. I previously had version 1.1.0 and see the same issue on version 1.1.4. I'm curious, could it be related to the current change in the code?

```diff
+  let regex = await nvim.eval('vimtex#re#deoplete')
-  let regex = await nvim.getVar('vimtex#re#deoplete')
```
